src/meshapi/views/query_api.py

```diff
@@ -18,128 +30,68 @@
-        auth=[],
-    ),
-)
-class QueryBuilding(generics.ListAPIView):
-    serializer_class = QueryFormSerializer
-    pagination_class = None
-    permission_classes = [permissions.LegacyMeshQueryPassword]
-
-    def get_queryset(self):
-        buildings = filter_model_on_query_params(
-            self.request.query_params,
-            Building,
-            BUILDING_FILTERS,
-        )
+class QueryFilterListAPIView(FilterRequiredListAPIView):
+    def get(self, request, *args, **kwargs):
+        maybe_auth = self.request.headers["Authorization"]
+        if maybe_auth != f"Bearer {os.environ.get('QUERY_PSK')}":
+            return Response({"detail": "Invalid Password"}, 403)
+        return super().get(request, *args, **kwargs)
 
-        output_qs = []
-        for building in buildings:
-            output_qs.extend(building.installs.all())
 
-        return output_qs
+class QueryMemberFilter(filters.FilterSet):
+    name = filters.CharFilter(field_name="member.name", lookup_expr="icontains")
+    email_address = filters.CharFilter(method="filter_on_all_emails")
+    phone_number = filters.CharFilter(field_name="member.phone_number", lookup_expr="icontains")
 
+    def filter_on_all_emails(self, queryset, name, value):
+        return queryset.filter(
+            Q(member__primary_email_address__icontains=value)
+            | Q(member__stripe_email_address__icontains=value)
+            | Q(member__additional_email_addresses__icontains=value)
+        )
 
-MEMBER_FILTERS = {
-    "email_address": None,
-}
+    class Meta:
+        model = Install
+        fields = []
 
 
-@extend_schema_view(
-    get=extend_schema(
-        tags=["Legacy Query Form"],
-        parameters=[query_form_password_param] + map_query_filters_to_param_annotations(MEMBER_FILTERS),
-        summary="Query & filter based on Member attributes. "
-        "Results are returned as flattened spreadsheet row style output",
-        auth=[],
-    ),
-)
-class QueryMember(generics.ListAPIView):
+@permission_classes([permissions.AllowAny])
+class QueryMember(QueryFilterListAPIView):
+    queryset = Install.objects.all().order_by("install_number")
     serializer_class = QueryFormSerializer
-    pagination_class = None
-    permission_classes = [permissions.LegacyMeshQueryPassword]
-
-    def get_queryset(self):
-        members = filter_model_on_query_params(
-            self.request.query_params,
-            Member,
-            MEMBER_FILTERS,
-            {
-                "email_address": "primary_email_address",
-            },
-        )
+    filterset_class = QueryMemberFilter
 
-        output_qs = []
-        for member in members:
-            output_qs.extend(member.installs.all())
 
-        return output_qs
+class QueryInstallFilter(filters.FilterSet):
+    network_number = filters.NumberFilter(field_name="node__network_number", lookup_expr="exact")
+    install_number = filters.NumberFilter(field_name="install_number", lookup_expr="exact")
```

Comment:
nit: this could be added to `Meta.fields` below since this is the default behavior I believe